pkg/operator/encryption/controllers/state_controller.go

```diff
@@ -103,6 +105,78 @@ func (c *stateController) sync() error {
 	return configError
 }
 
+type eventWithReason struct {
+	reason  string
+	message string
+}
+
+// eventsFromEncryptionConfigChanges return slice of event reasons with messages corresponding to a difference between current and desired encryption state.
+func eventsFromEncryptionConfigChanges(current, desired map[schema.GroupResource]state.GroupResourceState) []eventWithReason {
+	result := []eventWithReason{}
+	// handle removals from current first
+	for currentGroupResource := range current {
+		if _, exists := desired[currentGroupResource]; !exists {
+			result = append(result, eventWithReason{
+				reason:  "EncryptionResourceRemoved",
+				message: fmt.Sprintf("Resource %q was removed from encryption config", currentGroupResource),
+			})
+		}
+	}
+	for desiredGroupResource, desiredGroupResourceState := range desired {
+		currentGroupResource, exists := current[desiredGroupResource]
+		if !exists {
+			keyMessage := "with write key"
```

Comment:
do we want to put the key id in the msg? the key id has a special meaning.

Reply:
i can